Source/WebCore/page/ScreenOrientation.cpp

```diff
@@ -110,6 +110,12 @@ void ScreenOrientation::lock(LockType lockType, Ref<DeferredPromise>&& promise)
         promise->reject(Exception { SecurityError, "Only first party documents can lock the screen orientation"_s });
         return;
     }
+
+    if (document->page() && !document->page()->isVisible()) {
+        promise->reject(Exception { SecurityError, "Only visible documents can lock the screen orientation"_s });
+        return;
+    }
+
```

Comment:
Unrelated to this patch, but we could try to use DOMPromiseDeferred::settle in manager->lock().
Maybe by adding settle(std::optional<Exception>) to it?